admin/code/AdminRootController.php

```diff
@@ -4,15 +4,56 @@
  * @package framework
  * @subpackage admin
  */
-class AdminRootController extends Controller {
+class AdminRootController extends Controller implements TemplateGlobalProvider {
 
 	/**
 	 * @var string
 	 * @config
-	 * The url base that all LeftAndMain derived panels will live under
-	 * Won't automatically update the base route if you change this - that has to be done seperately
+	 *
+	 * Overrides {@link Director::$rules} config for the 'base' url for CMS administration areas
+	 * and the url base that all LeftAndMain derived panels will live under
 	 */
-	private static $url_base = 'admin';
+	private static $url_base;
+
+	/**
+	 * Convenience function to return the admin route config.
+	 * Returns the {@config url_base} override if set
+	 * otherwise falls back to the {@link Director::$rules}
+	 */
+	public static function get_admin_route() {
+		$override = Config::inst()->get('AdminRootController', 'url_base');
+
+		if ($override) {
+			return $override;
+		}
+		else{
+			$rules      = Config::inst()->get('Director', 'rules');
+			$adminRoute = array_search('AdminRootController', $rules);
```

Comment:
Please don't align assignments. Even though its not specifically pointed out in the conventions, we should follow implicit conventions here - the vast majority of existing assignments don't align.